llvm/lib/Target/AMDGPU/SIWholeQuadMode.cpp

```diff
@@ -1676,6 +1676,8 @@ bool SIWholeQuadMode::runOnMachineFunction(MachineFunction &MF) {
   if (!(GlobalFlags & (StateWQM | StateStrict)) && LowerToCopyInstrs.empty() &&
```

Comment:
I am now even more convinced that we should not bother with this fast path. We should just make sure that the general path is fast when there is no work to do.

Reply:
I'll run compile time testing on a version without it -- it's possible there really is little difference already.

Reply:
The only thing I can see that we _might_ be able to skip over in some cases is the two `for (auto BII : Blocks)` loops below.